### PR TITLE
Group dev

### PR DESCRIPTION
**Pull request by randallgray**
The submitted changes allow for Python3 compatibility.  This will allow you to run the driver under WeeWX 4.x.  Currently, it's been tested and functions with WeeWX 4.0.0b5.  

The sensors I'm using all rely on Python3 libraries which only run under 4.x and Thomas is no longer developing for 3.x so upgrading seems like a good idea anyhow...

One other thing to mention -- I have not yet verified the backward compatibility with Py 2.7.x.  My hope is that it will work with either.

Thank you!

### CODE REVIEW

**Review by jardiamj — CHANGES_REQUESTED** [review]
Hi, thanks for the PR.
I haven't done some needed clean up in this driver. This `get_average()` function is no longer needed neither does the `get_average_direction()` inside the `WindGauge` class; they can be safely removed.
I wrote those functions when I started writing the driver because, originally, it was averaging the wind speed and direction over a time interval. After asking in the forums, Thomas let me know that it was not recommended for a driver to be doing extra calculations because weeWX already has some Accumulator functions that do that. So, after experimenting a little bit I decided to change the driver to just sampling the wind direction at every interval but I didn't remove those functions even though they are not being used.
I doesn't hurt to leave the functions there, but if you indent the `get_average()` function you would also have to change line 303 from: 
`return get_average(data)`
to:
`return self.get_average(data)`
Because then the `get_average()` function would  belong to the `WindGauge` class.

File: byows_rpi.py [hunk 1]
```diff
@@ -147,7 +147,7 @@ def get_rainfall(self):
         self.reset_rainfall()
         return rainfall
 
-    def get_data(self):
+    def get_data(self, data):
         """ Generates data packets every time interval. """
         data = dict()
         anem_rotations = self.wind_gauge.wind_count / 2.0
```
Comment (jardiamj) [inline byows_rpi.py:153]: I don't understand the why of this change. The `data` variable used in the function is a dictionary that is initiated inside of it on line 152:
`data = dict()`
Then we populate the dictionary with each one of our observations. Here the `data` variable is being overwritten on line 152 by an empty dictionary as soon as it enters the function.
What error were you getting that drove you to make that change?

**Comment by randallgray** [comment]
Thank you Jardi for replying to my request.  I agree that the functions can
stay even though they are no longer being used -- I appreciate the logic of
your code and am glad it's still there to see and learn from.

Obviously, I didn't read that last part of the code close enough -- sorry
for suggesting something so stupid.

Just want to say thank you for what you've done so far -- it helped
tremendously having something to start with.

rg

**Review by jardiamj — COMMENTED** [review]
Thanks again for taking the time to submit this PR. No need to be sorry, I know the code that's why I spotted those things right away.
I agree in that the `get_average()` should be part of the `WindGauge` class, so if you fix the line 303 and the other comment I posted (line 50) I will merge your changes.
I need to go through the code and see what other things need to be cleaned up. Also, I don't have the hardware to test the code so if you can help improving it, that would be awesome.

**Pull request closed without merging**